readthedocs/config/config.py

```diff
@@ -582,6 +565,389 @@ def build_image(self):
         return self._config['build']['image']
 
 
+class BuildConfigV2(BuildConfigBase):
+
+    """Version 2 of the configuration file."""
+
+    version = '2'
+    valid_formats = ['htmlzip', 'pdf', 'epub']
+    valid_build_images = ['1.0', '2.0', '3.0', 'stable', 'latest']
+    default_build_image = 'latest'
+    valid_install_options = ['pip', 'setup.py']
+    valid_sphinx_builders = {
+        'sphinx': 'sphinx',
+        'htmldir': 'sphinx_htmldir',
+        'singlehtml': 'sphinx_singlehtml',
```

Comment:
I leave this values as plain `htmldir` rather than `sphinx_htmldir` to avoid redundancy in the `sphinx` key, but we always return the full name from the config object.

Reply:
should the first builder should be `html` then, instead of `sphinx`? I know our mapping has `sphinx` for the default html builder, but this is not clear, and should probably not be this way anyways. I think `{'html': 'sphinx_html'}` make more sense for this. Also, from the user perspective, I'd expect to fill in `sphinx.builder: html`.

Reply:
Yeah that makes more sense, I wasn't sure what to put there p: Changing

Reply:
Well, it looks like you followed our existing pattern, it's just that our existing pattern is sort of wrong also. Do we also need to update the logic for deciding on the builder?

Reply:
We don't need to update the logic here, but yeah when implementing the actual changes in the logic we need to modify that.

About the pattern, I'm not sure if I follow you hehe, do you mean changing the names from `sphinx` -> `html` there too? We can create an issue to track that.

Reply:
👍 that works